app/lib/mail/personalisation.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# This module is meant to be registered as an email interceptor on Action Mailer for developers
+# debugging and testing purposes.
+#
+# It adds the :subject_tags personalisation field:
+#    For non-production or test emails it adds a personalisation entry like this:
+#        subject_tags: "[local to:aaa@d1.md1,bbb@d2.md2]"
+#
+#    For production or test environments, the entry added is empty:
+#        subject_tags: ""
+#
+# Also, for non production or test environments, emails with a :subject personalisation field are modified to
+# preprend the app_name (from ENV['VCAP_APPLICATION'] || 'local') and the list of destination emails addresses
+# like this:
+#       subject: "[app_name to:aaa@d1.md1,bbb@d2.md2] original_subject_content"
+#
+module Mail
+  module Personalisation
+    class << self
+      def delivering_email(mail)
+        set_personalisation(mail, :subject_tags, subject_tags(mail))
+
+        # TODO: This enforces backward compatibility with existing emails and should be deleted as soon as all the emails
+        # are modified to use `subject_tags` instead
+        override_personalisation(mail, :subject) { |subject| new_subject(mail, subject) } if enabled?
```

Comment:
are we leaving this in for queued emails? as this only applies in test envs, and not sure if we still need it 🤔